Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1126 @@
+                    if (order == null)
+                    {
+                        // not our order, nothing else to do here
+                        return;
+                    }
+                }
+
+                var symbol = _symbolMapper.GetLeanSymbol(orderDetails.tradingSymbol, _symbolMapper.GetBrokerageSecurityType(orderDetails.tradingSymbol, orderDetails.exchange), orderDetails.exchange);
+                var fillPrice = decimal.Parse(orderDetails.filledPrice, NumberStyles.Float, CultureInfo.InvariantCulture);
+                var fillQuantity = decimal.Parse(orderDetails.filledQuantity, NumberStyles.Float, CultureInfo.InvariantCulture);
+                var updTime = DateTime.UtcNow;
+                var orderFee = new OrderFee(new CashAmount(CalculateBrokerageOrderFee(fillPrice * fillQuantity, order.Direction), Currencies.INR));
+
+                if (order.Direction == OrderDirection.Sell)
+                {
+                    fillQuantity = -1 * fillQuantity;
+                }
+                var status = OrderStatus.Filled;
+                if (fillQuantity != order.Quantity)
+                {
+                    decimal totalFillQuantity;
+                    _fills.TryGetValue(order.Id, out totalFillQuantity);
+                    totalFillQuantity += fillQuantity;
+                    _fills[order.Id] = totalFillQuantity;
+
+                    status = totalFillQuantity == order.Quantity
+                        ? OrderStatus.Filled
+                        : OrderStatus.PartiallyFilled;
+                }
+
+                var orderEvent = new OrderEvent
+                (
+                    order.Id, symbol, updTime, status,
+                    order.Direction, fillPrice, fillQuantity,
+                    orderFee, $"Samco Order Event {order.Direction}"
+                );
+
+                // if the order is closed, we no longer need it in the active order list
+                if (status == OrderStatus.Filled)
+                {
+                    Order outOrder;
+                    CachedOrderIDs.TryRemove(order.Id, out outOrder);
+                    decimal ignored;
+                    _fills.TryRemove(order.Id, out ignored);
+                    _pendingOrders.TryRemove(brokerId, out outOrder);
+                }
+
+                OnOrderEvent(orderEvent);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Lock the streaming processing while we're sending orders as sometimes they fill before the REST call returns.
+        /// </summary>
+        public void LockStream()
+        {
+            Log.Trace("SamcoBrokerage.LockStream(): Locking Stream");
+            _streamLocked = true;
+        }
+
+        /// <summary>
+        /// Unlock stream and process all backed up messages.
+        /// </summary>
+        public void UnlockStream()
```

Comment:
We should use the new `BrokerageConcurrentMessageHandler` for handling stream locking, instead of `UnlockStream()` `LockStream()` implementations